src/other/broadcasting.jl

```diff
@@ -113,9 +120,60 @@ function _copyto_helper!(dfcol::AbstractVector, bc::Base.Broadcast.Broadcasted,
     end
 end
 
+function Base.Broadcast.broadcast_unalias(dest::AbstractDataFrame, src)
+    for col in eachcol(dest)
+        src = Base.Broadcast.unalias(col, src)
+    end
+    src
+end
+
+function Base.Broadcast.broadcast_unalias(dest::AbstractDataFrame, src::AbstractDataFrame)
+    if size(dest, 2) != size(src, 2)
+        throw(ArgumentError("Dimension mismatch in broadcasting."))
+    end
+    # col2 can be checked from col1 point as we are writing broadcasting
+    # results from 1 to ncol
+    wascopied = false
+    for col1 in axes(dest, 2)
+        for col2 in col1:ncol(src)
+            dcol = dest[col1]
+            scol = src[col2]
+            if Base.mightalias(dcol, scol)
+                if src isa SubDataFrame
```

Comment:
Maybe we could allow `copy(::SubDataFrame, copycols=false)`, and have it return a `DataFrame` with `view` columns? Then this special-case wouldn't be needed, right?

Reply:
We have to special case it. The reason is that unaliasing should not change the type of broadcasted object. In theory someone might write a code that depends on the fact that `SubDataFrame` was passed on RHS (`isa` test for example). This is one of these "delicate things" that broadcasting does under the hood.

In short we if we take `SubDataFrame` we have to return `SubDataFrame` after unaliasing, but making sure that columns in the parent are unaliased.

Reply:
Ah, too bad. Then maybe just add a comment mentioning this requirement.